src/java.base/share/classes/sun/net/util/SocketExceptions.java

```diff
@@ -69,10 +70,19 @@ public static IOException of(IOException e, SocketAddress addr) {
     private static IOException ofInet(IOException e, InetSocketAddress addr) {
         int port = addr.getPort();
         String host = addr.getHostString();
+        InetAddress resolved = addr.getAddress();
         StringBuilder sb = new StringBuilder();
         sb.append(e.getMessage());
         sb.append(": ");
         sb.append(host);
+        if (resolved != null) {
+            String resolvedAddr = resolved.getHostAddress();
+            if (resolvedAddr != null && !resolvedAddr.equals(host)) {
+                sb.append('(');
+                sb.append(resolvedAddr);
+                sb.append(')');
```

Comment:
An alternative here might be to just use addr.toString() as it already handles both the resolved and unresolved cases and uses the most usual convention of hostname/ipaddress.

Reply:
That's a good idea. It even removes code instead of adding more. I've update the PR accordingly.